internal/providercache/dir_modify_test.go

```diff
@@ -0,0 +1,98 @@
+package providercache
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/apparentlymart/go-versions/versions"
+	"github.com/google/go-cmp/cmp"
+
+	"github.com/hashicorp/terraform/addrs"
+	"github.com/hashicorp/terraform/internal/getproviders"
+)
+
+func LinkFromOtherCache(t *testing.T) {
```

Comment:
Note to self to fix in a subsequent PR: this name doesn't have the `Test` prefix, so this test presumably isn't actually running?